purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1313,10 +1318,13 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
                 platformInfo,
                 !observerMode
             )
+            if (proxyURL != null) {
+                debugLog("Purchases is being configured using a proxy for RevenueCat")
+            }
             val backend = Backend(
                 apiKey,
                 Dispatcher(service),
-                HTTPClient(appConfig)
+                HTTPClient(appConfig, baseURL = URL(proxyURL))
```

Comment:
I don't think I know how to test this without making the backend property of Purchases internal. Any idea @aboedo ?

Reply:
there isn't a clean way to do it from here, at least none that I can think of. 
I think this is fine, but if you wanted to find a good fix, thoughts:
- maybe this is a sign that we're violating law of Demeter here by messing with the internals of the Backend? 
- possible fixes for "don't talk to strangers" violation: 
    1. the base URL could either be a part of appConfig (which feels reasonable to me)
    2. alternatively, you could pass the baseURL directly to the backend and it in turn can figure out whether it needs a new HTTPClient and how to set it up. You could have one constructor with what the class needs to start normally, and one with dependency injection (i.e.: passing in the HTTPClient) for testing purposes. 
    3. if the HTTPClient is meant to be singleton-ish, it could also be a property of Purchases, and be tested that way. 
Purchases is already doing a lot already, and we should try to relieve it from responsibilities, but this would be one that doesn't feel like a stretch to me. 

Any of those feel fine to me, although I'd personally lean towards 1., since it feels nice and clean, and is pretty simple because it keeps the current design.

Reply:
We are violating a lot of laws here. This static function is the worst for testing. 

I like the `AppConfig` solution, I wasn't sure the proxy would fit inside AppConfig, but it makes a lot of sense. Will change. Thanks for the thoughtful comment!